src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -1478,6 +1480,26 @@ void G1CollectedHeap::stop() {
   _cm_thread->stop();
 }
 
+class G1VCPUThreadClosure : public ThreadClosure {
+private:
+  volatile jlong _vtime = 0;
+
+public:
+  virtual void do_thread(Thread *thread) {
+    Atomic::add(&_vtime, os::thread_cpu_time(thread));
+  }
+  jlong vtime() { return _vtime; };
+};
+
+double G1CollectedHeap::elapsed_gc_vtime() {
+  G1VCPUThreadClosure cl;
+  _cr->threads_do(&cl);
+  _cm->threads_do(&cl);
+  _workers->threads_do(&cl);
```

Comment:
There is `CollectedHeap::gc_threads_do()` (in general, for all collectors) to iterate over GC threads.